### PR TITLE
OpenFOAM and scotch 64 bit easyblocks (REVIEW)

### PR DESCRIPTION
**Pull request by hajgato**

### CODE REVIEW

**Comment by wpoely86** [comment]
lgtm

**Comment by boegel** [comment]
Is this still WIP?

**Comment by hajgato** [comment]
@boegel: just finished the build. So now it is in REVIEW. Anyway, `OpenFOAM 3.0.0` 64 bit integer version does not work with `METIS`. `SCOTCH` works with the 64bitint bit version (in the  PR https://github.com/hpcugent/easybuild-easyconfigs/pull/2142)

**Comment by boegel** [comment]
@wpoely86: please review?

**Comment by wpoely86** [comment]
looks good

@hajgato maybe add a warning/error if `METIS` is loaded and `i8` is set?

**Comment by hajgato** [comment]
@wpoely86 I filed a bug report for `OpenFOAM` we have now a patch for 64 bit `METIS` and the future versions will support 64bit `METIS`. I do not know how could I check whether a patch exists for the use of 64bit `METIS`

File: easybuild/easyblocks/o/openfoam.py [hunk 1]
```diff
@@ -192,6 +197,11 @@ def configure_step(self):
         # parallel build spec
         env.setvar("WM_NCOMPPROCS", str(self.cfg['parallel']))
 
+        # OpenFOAM >= 3.0.0 can use 64 bit integers
+        if 'extend' not in self.name.lower() and LooseVersion(self.version) >= LooseVersion('3.0'):
+            if self.toolchain.options['i8']:
+                env.setvar("WM_LABEL_SIZE", '64')
```
Comment (boegel) [inline easybuild/easyblocks/o/openfoam.py]: no `else:` here?

File: easybuild/easyblocks/o/openfoam.py [hunk 2]
```diff
@@ -260,7 +270,16 @@ def install_step(self):
     def sanity_check_step(self):
         """Custom sanity check for OpenFOAM"""
 
-        psubdir = "linux64%sDPOpt" % self.wm_compiler
+        # OpenFOAM >= 3.0.0 can use 64 bit integers
+        if 'extend' not in self.name.lower() and LooseVersion(self.version) >= LooseVersion('3.0'):
+            if self.toolchain.options['i8']:
+                int_size='Int64'
```
Comment (boegel) [inline easybuild/easyblocks/o/openfoam.py]: style remark: spaces around ` = `

File: easybuild/easyblocks/o/openfoam.py [hunk 3]
```diff
@@ -260,7 +270,16 @@ def install_step(self):
...
+            else:
+                int_size='Int32'
```
Comment (boegel) [inline easybuild/easyblocks/o/openfoam.py]: style remark: spaces around ` = `

File: easybuild/easyblocks/o/openfoam.py [hunk 4]
```diff
@@ -260,7 +270,16 @@ def install_step(self):
...
+        else:
+            int_size=''
```
Comment (boegel) [inline easybuild/easyblocks/o/openfoam.py]: style remark: spaces around `=`

File: easybuild/easyblocks/o/openfoam.py (hunk 1 above)
Comment (boegel) [inline easybuild/easyblocks/o/openfoam.py]: (see below, where you have and `else:` that sets it to `32`)

File: easybuild/easyblocks/o/openfoam.py [hunk 5]
```diff
@@ -128,7 +129,11 @@ def configure_step(self):
         for script in [os.path.join(self.builddir, self.openfoamdir, x) for x in ['etc/bashrc', 'etc/cshrc']]:
             self.log.debug("Patching out hardcoded $WM_* env vars in %s", script)
             for line in fileinput.input(script, inplace=1, backup='.orig.eb'):
-                for env_var in ['WM_COMPILER', 'WM_MPLIB', 'WM_THIRD_PARTY_DIR']:
+                WM_env_var = ['WM_COMPILER', 'WM_MPLIB', 'WM_THIRD_PARTY_DIR']
+                # OpenFOAM >= 3.0.0 can use 64 bit integers
+                if 'extend' not in self.name.lower() and LooseVersion(self.version) >= LooseVersion('3.0'):
+                    WM_env_var += ['WM_LABEL_SIZE']
```
Comment (boegel) [inline easybuild/easyblocks/o/openfoam.py]: style:

```python
WM_env_var.append('WM_LABEL_SIZE')
```

File: easybuild/easyblocks/o/openfoam.py (hunk 1 above)
Comment (hajgato) [inline easybuild/easyblocks/o/openfoam.py]: @boegel If we do not set, it defaults to 32 bit. But you are right, I should be consistent.

**Comment by hajgato** [comment]
@boegel style corrected

**Comment by boegel** [comment]
Jenkins: test this please

**Comment by boegel** [comment]
Going in, thanks @hajgato!

cc @wpoely86

**Pull request merged**